draft/XXXX-upload-system/README.md

```diff
@@ -0,0 +1,126 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-07-08
+Last-Modified: 2020-07-08
+---
+
+# RFC XXXX: Advanced and Robust Upload System
+
+Uploading a large dataset can be both timeconsuming and problematic. This RFC proposes a method for handling uploads that both facilitates ease of use, as well as reliability and fault tolerance. Uploading data is one of the most important parts of this system, and as such should be robust and featured.
+
+## Motivation
+
+Our current system of dealing with uploads is very simple and unfeatured. We send the data through a single request, with no track of progress or anything. This exposes a couple of key issues:
+
+* If the upload takes too long, the connection times out (due to a current gunicorn setting)
+* If the connection is interrupted, the upload must start over from the beginning (very frustrating on large datasets)
+* There's no progress/indication to the user. This can be implemented on the _client_ by simply tracking the HTTP request progress, but as a matter of exposing this functionality through our API, it does not exist.
+
+## Proposals
+
+### Chunked Uploads
+Instead of dealing with a single request to upload data, this data could instead be sent in chunks, allowing for many smaller HTTP requests. This would allow the following things:
+* Retain a timeout on requests, while also allowing large datasets to be uploaded with ease.
+* Fault tolerance and upload pausing/resuming
+* Progress tracking
+
+This would be accomplished through the following additions.
+
+### Hidden Tables/Networks
+To allow for uploaded chunks to be stored somewhere, when starting an upload, the desired table/network would be created in a "hidden" mode. This would prevent the table from being returned by any table listing endpoints. This table should be assumed to be volatile, as data is still being uploaded to it in this state. Whether a table is hidden or not would be tracked by a `_meta` collection on each workspace. This would contain, among other things, the information of any hidden tables in that workspace. Each document in this collection will have a structure like so:
```

Comment:
I think we could use a term besides "hidden", as this implies something about the _intended_ visibility of the table. Really this flag is meant to explain that the table is in an intermediate state. We choose to hide such tables, but that doesn't mean the table itself is "hidden".